packages/edit-post/src/components/editor-initialization/index.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useContext } from '@wordpress/element';
+import { RegistryContext } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import {
+	blockSelectionListener,
+	adjustSidebarListener,
+	viewPostLinkUpdateListener,
+} from './listeners';
```

Comment:
Rather than listener functions called by a single subscription, could we have each of these behaviors implemented in their own components and re-use existing tools like `withSelect` to feed data (to which updates occur using either `componentDidUpdate` or a hooks equivalent)?

Examples:

- https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/multi-select-scroll-into-view/index.js
- https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/preserve-scroll-in-reorder/index.js
- https://github.com/WordPress/gutenberg/blob/master/packages/edit-post/src/components/browser-url/index.js
- https://github.com/WordPress/gutenberg/blob/master/packages/edit-post/src/components/fullscreen-mode/index.js
- https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/components/autosave-monitor/index.js
- https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/components/unsaved-changes-warning/index.js

Reply:
We could, I'm keen on understanding what advantages this provides over the approach in this pull (there's probably something I'm missing because its not immediately obvious to me :) )?

Reply:
So one benefit that _is_ apparent is it helps keep things more singular in purpose?

Reply:
> So one benefit that _is_ apparent is it helps keep things more singular in purpose?

Yes, I think it helps keep the behaviors single-focused. It also allows us to reuse existing, well-understood patterns (`withSelect`) than to create custom subscriber implementations.

Reply:
This has been refactored to now use custom hooks.